src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -277,6 +280,28 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
      */
     void setTextOnSymbolTextFormat( const QgsTextFormat &format ) { mTextOnSymbolTextFormat = format; }
 
+    /**
+     * Returns the legend label of the layer or symbol.
+     * \since QGIS 3.8
+     */
+    QString getCurrentLabel() const;
```

Comment:
What's the difference between this one and `evaluateLabel()` exactly?

Sidenote: we don't use `get` prefixes

Reply:
This one is used to obtain the label assigned to the symbol by checking the distinct places where it can be store. The evaluate label evaluate either this label or the expression set in the layertreelayer.

This is a way to fetch the string assigned to the symbol and output it in the context and prevent having to do the same tasks elsewhere in the code.

Reply:
Would it make sense to rename it to `labelExpression()`?

Reply:
I'm going for symbolLabel, is that ok?

Reply:
Is it not an expression?

Reply:
No it is used to retrieve either the symbol label, that may or may not contains expression if the user modified it.

Reply:
For me the "that may contain expression" part is what should be communicated through the method signature in some way. I.e. let's make it clear to a user that this is not "ready for usage and presentation to the user".

Reply:
This returns what is currently the symbol label, if the users changes it, this is what will be provided, otherwise the default label of a symbol will be provided. The expression bound to this function is called symbol_label. This should be clear enough and prevent confusion.